### PR TITLE
[WIP] MIOPEN Conv Ops Refactoring

### PR DESCRIPTION
**Pull request by petrex**
- Enable miopen DepthWise conv ops
- Drop dilation constraint on miopen group conv

### CODE REVIEW

**Comment by petrex** [comment]
@bddppq

File: caffe2/operators/hip/conv_op_miopen.hip [hunk 1]
```diff
@@ -49,17 +49,15 @@ class MIOPENConvOpBase : public ConvPoolOpBase<HIPContext> {
     MIOPEN_ENFORCE(miopenCreateTensorDescriptor(&top_desc_for_bias_));
     MIOPEN_ENFORCE(miopenCreateConvolutionDescriptor(&conv_desc_));
 
-    if (group_ > 1) {
+    if ((operator_def.type().substr(0, 15) == "DepthWiseConv") ||
```
Comment (bddppq) [inline caffe2/operators/hip/conv_op_miopen.hip]: We are moving away from using OperatorDef. You can instead add a template parameter 
```
typename <bool DepthWise>
class MIOPENConvOpBase : ... 
```
and then set `mode_` based on `DepthWise`

**Comment by bddppq** [comment]
ping @petrex

**Comment by petrex** [comment]
Thanks for the feedback. will address the comment in later PRs.

**Comment by petrex** [comment]
@bddppq We have consolidated the design of miopen Conv. The enum `miopenDepthwise /miopenGroupConv` are now deprecated. Closing this PR.

**Pull request closed without merging**